tests/integration/cattletest/core/test_svc_discovery.py

```diff
@@ -190,6 +190,7 @@ def test_activate_single_service(client, context, super_client):
     assert svc.launchConfig.healthCheck.port == 200
     assert svc.metadata == metadata
     assert svc.launchConfig.version == '0'
+    assert svc.launchConfig.requestedHostId == host.id
```

Comment:
@prachidamle @vincent99 very much wants the embedded types to have a type field on them.  So can you make it such that the below works.  Make sure that if the embedded type already has a field named "type" we don't override it.

```python
assert assert svc.launchConfig.type == 'launchConfig'
````